kube-derive/src/lib.rs

```diff
@@ -214,5 +180,40 @@ use custom_resource::CustomResource;
 /// [`k8s_openapi::Resource`]: https://docs.rs/k8s-openapi/*/k8s_openapi/trait.Resource.html
 #[proc_macro_derive(CustomResource, attributes(kube))]
 pub fn derive_custom_resource(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
-    run_custom_derive::<CustomResource>(input)
+    let input = proc_macro2::TokenStream::from(input);
+    let tokens = input.clone();
+
+    let derive_input: DeriveInput = match syn::parse2(input) {
+        Err(err) => return err.to_compile_error().into(),
+        Ok(di) => di,
+    };
+    // Limit derive to structs
+    match derive_input.data {
+        Data::Struct(_) => {}
+        _ => {
+            return syn::Error::new_spanned(
+                &derive_input.ident,
+                r#"Enums or Unions can not #[derive(CustomResource)"#,
+            )
+            .to_compile_error()
+            .into()
+        }
+    }
+
+    let kube_attrs = match KubeAttrs::from_derive_input(&derive_input) {
+        Err(err) => return err.write_errors().into(),
+        Ok(attrs) => attrs,
+    };
+
+    let output = match custom_resource::derive(derive_input, kube_attrs) {
+        Ok(out) => match syn::parse2(out) {
```

Comment:
I'm not sure if this `syn::parse2` is necessary.

The original is:
https://github.com/clux/kube-rs/blob/6b9ce1f41d5cf3fcf8c71e920a4c0fee2ab66252/kube-derive/src/custom_resource.rs#L491-L493

where `output` is `proc_macro2::TokenStream`